doc/Language/compilation.pod6

```diff
@@ -0,0 +1,295 @@
+=begin pod :kind("Language") :subkind("Language") :category("tutorial")
+
+=TITLE CompUnits and where to find them
+
+=SUBTITLE How and when Raku modules are compiled, where they are stored, and how to access them in compiled form.
+
+=head1 Overview
+
+Programs in Raku, as a member of the Perl language family, tend at the top level to be more at the interpreted
+end of the interpreted-compiled spectrum. In this tutorial, an 'interpreted' program means that the source code,
+namely the human-readable text such as C<say 'hello world';> is immediately processed by the C<Raku> program into code
+that can be executed by the computer, with any intermediate stages being stored in memory.
+
+A compiled program, by contrast, is one where the human readable source is first processed into machine-executable code
+and some form of this code is stored 'on disc'. In order to execute the program, the machine-readable version is loaded
+into memory and then run by the computer.
+
+Both compiled and interpreted forms have advantages. Briefly, interpreted programs can be 'whipped up' quickly and
+the source changed quickly. Compiled programs can be complex and take a significant time to pre-process into machine-readable
+code, but then running them is much faster for a user, who only 'sees' the loading and running time, not the compilation
+time.
+
+C<Raku> has both paradigms. At the B<top level> a Raku program is interpreted, but if code that is separated out into a
+Module will be compiled and the preprocessed version is then loaded when necessary. In practice, Modules that have been
+written by the community will only need to be pre-compiled once by a user when they are 'installed', for example by a
+Module manager such as C<zef>. Then they can be C<use'd>  by a developer in her own program. The effect is to make C<Raku>
+top level programs run quickly.
+
+One of the great strengths of the C<Perl> family of languages was the ability to integrate a whole ecosystem of modules
+written by competent programmers into a small program. This strength was widely copied and is now the norm for all
+languages. C<Raku> takes integration even further, making it relatively easy for C<Raku> programs to incorporate system
+libraries written in other languages into C<Raku> programs, see L<Native Call|Language/nativecall>.
+
```

Comment:
This is probably not relevant for this tutorial...

Reply:
I was trying to show that Raku has both compilation and interpreting paradigms, and also the ability to integrate with system libraries is not something I have found in other languages.

I think that the concept that the top level program is in memory, but modules are pre-compiled (by default) should be highlighted.
I appreciate my way of saying this may not be optimal.